### PR TITLE
[FEAT BRANCH] Add `argilla-sdk` project

### LINKED ISSUE #4895: include CI/CD for docs

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

This PR setup the argilla-sdk project. This PR only copy the code base from the [argilla-python repository](https://github.com/argilla-io/argilla-python). ~~No changes in package definition or docs have been applied yet.~~ UPDATED: The only change is [this line](https://github.com/argilla-io/argilla/blob/feat/add-new-sdk-project/argilla-sdk/src/argilla_sdk/_helpers/_uuid.py#L61) which raise an ruff error when applying commit hooks. 

## Update

The code on this PR won't work with the server version on `feat/v2.0.0` until we resolve SDK tasks for the issue https://github.com/argilla-io/argilla/issues/4773. For the sake of the review complexity, I would prefer to tackle this migration as a separate PR.

After those changes, the new SDK will still be fully compatible with argilla server 1.29.0. 

Closes https://github.com/argilla-io/argilla/issues/4888
Closes https://github.com/argilla-io/argilla/issues/4895

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactor (change restructuring the codebase without changing functionality)
- [ ] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

(Please describe the tests that you ran to verify your changes. And ideally, reference `tests`)

- [ ] Test A
- [ ] Test B

**Checklist**

- [ ] I added relevant documentation
- [ ] I followed the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I filled out [the contributor form](https://tally.so/r/n9XrxK) (see text above)
- [ ] I have added relevant notes to the `CHANGELOG.md` file (See https://keepachangelog.com/)

### CODE REVIEW

File: argilla-sdk/mkdocs.yml
Comment (frascuchon) [inline argilla-sdk/mkdocs.yml:11]: I saw this line in the latest sync. Should this affect the current project setup? In that case, should we comment until we have merged the project into the default/develop/main one? cc @davidberenstein1957 @burtenshaw

**Comment by frascuchon** [comment]
I think we should review also some changes in docs: https://github.com/argilla-io/argilla/actions/runs/9352932120/job/25742223903?pr=4891#step:10:39 cc @davidberenstein1957  @sdiazlor

**Review by sdiazlor — APPROVED** [review]
Regarding the docs, looks like they are all.

**Pull request merged**
